tests/drop_counters/test_drop_counters.py

```diff
@@ -247,17 +323,21 @@ def base_verification(discard_group, pkt, ptfadapter, duthost, combined_counter,
         pytest.fail("Incorrect 'discard_group' specified. Supported values: 'L2' or 'L3'")
 
 
-def test_equal_smac_dmac_drop(ptfadapter, duthost, setup, tx_dut_ports, pkt_fields):
+def test_equal_smac_dmac_drop(ptfadapter, fanout, duthost, setup, tx_dut_ports, pkt_fields):
     """
     @summary: Verify that packet with equal SMAC and DMAC is dropped and L2 drop cunter incremented
     """
     dut_iface, ptf_tx_port_id, dst_mac, src_mac = get_test_ports_info(ptfadapter, duthost, setup, tx_dut_ports)
+    src_mac = "00:00:00:00:00:11"
 
     log_pkt_params(dut_iface, dst_mac, dst_mac, pkt_fields["ip_dst"], pkt_fields["ip_src"])
 
+    # Prepare openflow rule
+    fanout.exec_template(template_path=os.path.join(os.path.dirname(__file__), "mlnx_fanout_add_rule_eq_mac.j2"), match_mac=src_mac, set_mac=dst_mac)
```

Comment:
This is hardcoded for mellanox fanout only. How can users add handling for their own fanouts without rewriting all the code in this PR ?

Reply:
Changed approach, now added possibility to define vendor specific module and use it inside 'fanouthost' fixture and to update/restore config.